src/core/choroplethFeature.js

```diff
@@ -0,0 +1,245 @@
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth = function (arg1, arg2) {
+    if (arg1 === undefined) {
+      return m_choropleth;
+    }
+    if (typeof arg1 === 'string' && arg2 === undefined) {
+      return m_choropleth[arg1];
+    }
+    if (arg2 === undefined) {
+      var choropleth = $.extend(
+        {},
+        m_choropleth,
+        arg1
+      );
+      m_choropleth = choropleth;
+    } else {
+      m_choropleth[arg1] = arg2; //if you pass in accessor for prop
+    }
+    m_this.modified();
+    return m_this;
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A uniform getter that always returns a function even for constant values.
+   * If undefined input, return all the choropleth values as an object.
+   *
+   * @param {string|undefined} key
+   * @return {function}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this.choropleth.get = function (key) {
+    if (key === undefined) {
+      var all = {}, k;
+      for (k in m_choropleth) {
+        if (m_choropleth.hasOwnProperty(k)) {
+          all[k] = m_this.choropleth.get(k);
+        }
+      }
+      return all;
+    }
+    return geo.util.ensureFunction(m_choropleth[key]);
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds a polygon feature to the current layer.
+   *
+   * @param {array} coordinateArray
+   * @param {geo.color} fillColor
+   * @return {geo.feature}
+   */
+  ////////////////////////////////////////////////////////////////////////////
+  this._addPolygonFeature = function(coordinateArray, fillColor){
+    return m_this.layer()
+      .createFeature('polygon', {
+      })
+      .data([{
+        type: 'Polygon',
+        coordinates: coordinateArray
+      }])
+      .polygon(function (d) {
+        return {
+          'outer': d.coordinates[0],
+          'inner': d.coordinates[1] // undefined but ok
+        };                              
+      })
+      .position(function(d){
+        return {
+          x: d[0],
+          y: d[1]
+        };
+      })
+      .style({
+        'fillColor': fillColor
+      });
+  };
+
+  ////////////////////////////////////////////////////////////////////////////
+  /**
+   * A method that adds polygons from a given feature to the current layer.
+   *
+   * @param {} geoJsonFeature
+   * @param geo.color
+   * @return [{geo.feature}] 
+   */
+  ////////////////////////////////////////////////////////////////////////////          
+  this._featureToPolygons = function (feature, fillValue) {
+
+    var arrayOfPolygonFeatures = [];
+    
+    if (feature.geometry.type === 'Polygon'){
+      arrayOfPolygonFeatures = [
+        m_this
+          ._addPolygonFeature(feature.geometry.coordinates, fillValue)
+      ];
+      
+    }
+    if (feature.geometry.type === 'MultiPolygon') {
```

Comment:
> This defines a "feature class" to be a "collection of geographic
> features".  Unless I'm missing something, that means a "point feature
> class" on some map would be a bunch of "point features".  I am just
> pointing out that we're not using the word "feature" in GeoJS consistently
> either with common usage in English, or even with this GIS definition.

pointFeature is a class in GeoJS too. I think its verbose to name it pointFeatureGroup. I believe its just a matter of understanding GIS context and then it would make sense. I names classes keeping in mind that GIS folks would expect.